mailpoet/tests/integration/Subscribers/SubscriberSegmentRepositoryTest.php

```diff
@@ -0,0 +1,82 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Subscribers;
+
+use MailPoet\Entities\SubscriberEntity;
+use MailPoet\Entities\SubscriberSegmentEntity;
+use MailPoet\Test\DataFactories\Segment;
+use MailPoet\Test\DataFactories\Subscriber;
+
+class SubscriberSegmentRepositoryTest extends \MailPoetTest {
+
+  /** @var SubscriberSegmentRepository */
+  private $testee;
+
+  public function _before() {
+    $this->testee = $this->diContainer->get(SubscriberSegmentRepository::class);
+  }
+
+  public function testResetIsWorking() {
+    $subscriber = (new Subscriber())->create();
+    $segment1 = (new Segment())->create();
+    $segment2 = (new Segment())->create();
+    $segment3 = (new Segment())->create();
+    $segment4 = (new Segment())->create();
+
+    $this->testee->resetSubscriptions($subscriber, [$segment1, $segment2, $segment3, $segment4]);
+    $subscribedSegments = $subscriber->getSubscriberSegments(SubscriberEntity::STATUS_SUBSCRIBED);
+    $unsubscribedSegments = $subscriber->getSubscriberSegments(SubscriberEntity::STATUS_UNSUBSCRIBED);
+    $subscribedSegmentIds = array_map(
+      function(SubscriberSegmentEntity $entity): ?int {
+        return $entity->getSegment() ? $entity->getSegment()->getId() : null;
+      },
+      $subscribedSegments->toArray()
```

Comment:
This could be simplified by calling [the map method](https://www.doctrine-project.org/projects/doctrine-collections/en/stable/index.html#map) on the collection. Nevertheless, it is not a big deal, and I consider it optional.

What I would like to change here is to avoid code repetition because there is identical code three times. We could create a private method in this test class or a public method in the SubscriberEntity class.

Reply:
3cce6bca1